eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/implementation/ActiveClientTokenManager.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs.implementation;
+
+import com.azure.core.amqp.CBSNode;
+import com.azure.core.amqp.exception.AmqpResponseCode;
+import com.azure.core.implementation.logging.ServiceLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Manages the re-authorisation of the client to the token audience against the CBS node.
+ */
+class ActiveClientTokenManager implements Closeable {
+    private final ServiceLogger logger = new ServiceLogger(ActiveClientTokenManager.class);
+    private final AtomicBoolean hasScheduled = new AtomicBoolean();
+    private final Mono<CBSNode> cbsNode;
+    private final String tokenAudience;
+    private final Duration tokenValidity;
+    private final Duration refreshInterval;
+    private Timer timer;
```

Comment:
`timer` gets left out of the `final` party?

Reply:
🤣 `timer`'s invite was sent via snail mail.